repos/system_upgrade/common/actors/luksscanner/libraries/luksdump_parser.py

```diff
@@ -0,0 +1,195 @@
+class LuksDumpParser(object):
+    """
+    Class for parsing "cryptsetup luksDump" output. Given a list of lines, it
+    generates a dictionary representing the dump.
+    """
+
+    class Node(object):
+        """
+        Helper class, every line is represented as a node. The node depth is
+        based on the indentation of the line. A dictionary is produced after
+        all lines are inserted.
+        """
+
+        def __init__(self, indented_line):
+            self.children = []
+            self.level = len(indented_line) - len(indented_line.lstrip())
+            self.text = indented_line.strip()
+
+        def add_children(self, nodes):
+            childlevel = nodes[0].level
```

Comment:
``nodes[0]`` asumes non-empty list. This is probably not a problem, since we assume non-empty output from the command, but this fails otherwise

Reply:
I see your point. seems to me ok as to my understanding it should by always non-empty and I guess we could have a different problems somewhere else otherwise. I am thinking to raise an exception to catch the error and act properly, but otoh I do not know what else to do and what error msg to provide to customer. As it's more hypothetic, let's just add a comment and see if it ever appears - as in both cases, I would stop the upgrade raising an error anyway right now.

Reply:
thinking about it twice. I do not know whether to use hard blocker (raising an error anyway) or whether to just log a warning and return from the function. let's add for now a comment note and discuss it later.

Reply:
hmmm..interesting. why it's not marked as obsoleted when I changed this part of the code? :eyes:

Reply:
anyway. note added.